internal/cmd/genop/genop.go

```diff
@@ -63,23 +63,73 @@ func {{$name}}(n *node) {
 			{{else}}
 			v1 := genValueInt(c1)
 			{{end -}}
+			{{- if (eq $name "add")}}
+			if n.typ.cat != interfaceT || len(n.typ.field) > 0 {
+				n.exec = func(f *frame) bltn {
+					_, j := v1(f)
+					dest(f).SetInt(i + j)
+					return next
+				}
+				return
+			}
+			var valf func(sum int64) reflect.Value
+			switch typ.Kind() {
+			case reflect.Int:
+				valf = func(sum int64) reflect.Value { return reflect.ValueOf(int(sum)) }
+			case reflect.Int32:
+				valf = func(sum int64) reflect.Value { return reflect.ValueOf(int32(sum)) }
+			default: // int64
+				valf = func(sum int64) reflect.Value { return reflect.ValueOf(sum) }
+			}
+			n.exec = func(f *frame) bltn {
+				_, j := v1(f)
+				dest(f).Set(valf(i + j))
+				return next
+			}
+			{{else -}}
 			n.exec = func(f *frame) bltn {
 				_, j := v1(f)
 				dest(f).SetInt(i {{$op.Name}} j)
 				return next
 			}
+			{{end -}}
 		case c1.rval.IsValid():
 			v0 := genValueInt(c0)
 			{{- if $op.Shift}}
 			j := vUint(c1.rval)
 			{{else}}
 			j := vInt(c1.rval)
 			{{end -}}
+			{{- if (eq $name "add")}}
+			var valf func(sum int64) reflect.Value
+			switch typ.Kind() {
```

Comment:
idem as line 76: add a TODO for missing types

Reply:
done